Grigore-George-Mihai/sandbox_playground
# Add elasticsearch integration

## Summary
- This PR integrates Elasticsearch into the Post model using the searchkick gem. 
- It sets up full-text search capabilities on the title field and ensures that posts are reindexed after creation and updates.

## Related Issue(s)
- No

## Changes

## Description of Changes
- Elasticsearch Integration: Integrated the searchkick gem to enable full-text search on the Post model.
- Post Model Refactor: Added the searchkick configuration to the Post model to index the title field.

## How to Test
Steps to test the changes introduced in this PR (e.g., commands to run, environment setup, etc.).

1. Development Testing: Start your Rails server and test the search functionality in development by creating new posts and verifying that they are searchable by title after creation and updates.
2. Elasticsearch Verification: Ensure Elasticsearch is running, and check the indexed data by querying directly via the terminal or the app.
3. Run and check Rspec

## Screenshots (Optional)
If applicable, include screenshots to help explain your changes visually.

## Checklist
- [x] I have run `rubocop` and fixed any linter warnings.
- [x] I have run the test suite to ensure everything is passing.

## Additional Context (Optional)
Any other relevant information or context.
